src/style/styled_content.rs

```diff
@@ -85,13 +104,16 @@ impl<D: Display + Clone> Display for StyledContent<D> {
             reset = true;
         }
 
-        for attr in self.style.attributes.iter() {
+        for attr in &self.style.attributes {
             queue!(f, SetAttribute(*attr)).map_err(|_| fmt::Error)?;
             reset = true;
         }
 
-        fmt::Display::fmt(&self.content, f)?;
+        self.content.fmt(f)?;
 
+        // TODO: There are specific command sequences for "reset forground
+        // color (39m)" and "reset background color (49m)"; consider using
```

Comment:
Actually I think we should not reset to the default for or background. Instead, we probably want to undo the attributes and colors of this `StyledContent`.

Reply:
I agree. I looked into doing something like this many years ago. Unfortunately, because the attributes aren't stackful (that is, you can't "pop" a background color and restore the previous one), the only way to do this correctly would be to query the style before initiating the writes, so that we can restore it afterwards.

A *mostly* correct way to do this would to require our `QueuableCommand` instances to be wrapped in a struct, which can then manually keep track of the "style stack". This would be correct so long as is only one instance of this struct, and that the end user doesn't insert their own control sequences into the stream.

Reply:
However, solving that problem requires more design work, and *any* solution we come up with will be backwards incompatible, so I figured I'd leave the comment there for now, rather than creating a fix that we'd just have to change again down the line.

Reply:
There is an easier solution, like we have in the examples. We store both the enable and disable attributes in a tuple style array.

Attrs = [(BoldOff, BoldOn)]

We can query in that array for the attribute and take the opposite. This is similar to what cursive does. I do this in the terminal library as well.

Reply:
I'm curious to see, which example file does it?

Reply:
https://github.com/crossterm-rs/crossterm/blob/master/examples/interactive-demo/src/test/attribute.rs#L7

Reply:
Oh, that's not the part I'm worried about. The part I'm worried about is tracking WHICH state to restore after you finish writing a styled content. You can't just unconditionally invert.

For example, consider this pseudocode:

```
WriteUnderlined("abc", WriteUnderlined("1234"), "xyz")
```

If the inner `WriteUnderlined` calls `UnsetUnderline` after printing 1234, then "xyz" *won't* be underlined, even though it should be
```